test/e2e-go/cli/goal/expect/goalExpectCommon.exp

```diff
@@ -316,6 +316,162 @@ proc ::AlgorandGoal::WaitForAccountBalance { WALLET_NAME ACCOUNT_ADDRESS EXPECTE
     return $ACCOUNT_BALANCE
 }
 
+# Create an asset
+proc ::AlgorandGoal::AssetCreate { CREATOR WALLET_NAME WALLET_PASSWORD TOTAL_SUPPLY UNIT_NAME TEST_PRIMARY_NODE_DIR } {
+    set timeout 40
+    if { [ catch {
+        spawn goal asset create -d $TEST_PRIMARY_NODE_DIR -w $WALLET_NAME --creator $CREATOR --total $TOTAL_SUPPLY --unitname $UNIT_NAME
+        expect {
+            timeout { ::AlgorandGoal::Abort "Failed to create asset"  }
+	    "Please enter the password for wallet '$WALLET_NAME':" { send "$WALLET_PASSWORD\r" }
+        }
+    } EXCEPTION ] } {
+       ::AlgorandGoal::Abort "ERROR in AssetCreate: $EXCEPTION"
+    }
+}
+
+# Transfer asset
+proc ::AlgorandGoal::AssetTransfer { WALLET_NAME WALLET_PASSWORD FROM_ADDR TO_ADDR ASSET_ID ASSET_AMOUNT TEST_PRIMARY_NODE_DIR} {
+    if { [ catch {
+        spawn goal asset send -d $TEST_PRIMARY_NODE_DIR -w $WALLET_NAME --from $FROM_ADDR --to $TO_ADDR --assetid $ASSET_ID --amount $ASSET_AMOUNT 
+        expect {
+            timeout { ::AlgorandGoal::Abort "Failed to send asset"  }
+	    "Please enter the password for wallet '$WALLET_NAME':" { send "$WALLET_PASSWORD\r" }
+        }
+    } EXCEPTION ] } {
+       ::AlgorandGoal::Abort "ERROR in AsssetTransfer: $EXCEPTION"
+    }
+}
+
+# write asset transfer to a txn file
+proc ::AlgorandGoal::CreateAssetTransfer { FROM_ADDR TO_ADDR ASSET_ID ASSET_AMOUNT TEST_PRIMARY_NODE_DIR TXN_OUTPUT} {
+    if { [ catch {
+        spawn goal asset send -d $TEST_PRIMARY_NODE_DIR --from $FROM_ADDR --to $TO_ADDR --assetid $ASSET_ID --amount $ASSET_AMOUNT -o $TXN_OUTPUT         
+	expect {
+            timeout { ::AlgorandGoal::Abort "Failed to create asset transfer transaction"  }
+	    close
+        }
+    } EXCEPTION ] } {
+       ::AlgorandGoal::Abort "ERROR in CreateAsssetTransfer: $EXCEPTION"
+    }
+}
+
+# Get asset id
+proc ::AlgorandGoal::AssetLookup { CREATOR UNIT_NAME TEST_PRIMARY_NODE_DIR } {
+    set timeout 10
+    if { [ catch {
+	set ASSET_ID "NOT SET"
+        spawn goal asset info -d $TEST_PRIMARY_NODE_DIR --creator $CREATOR  --asset $UNIT_NAME
+        expect {
+            timeout { ::AlgorandGoal::Abort "Asset lookup timeout"  }
+            -re {Asset ID:\s+([0-9]+)} {set ASSET_ID $expect_out(1,string); close }
+        }
+    } EXCEPTION ] } {
+       ::AlgorandGoal::Abort "ERROR in AssetLookup: $EXCEPTION"
+    }
+    return $ASSET_ID
+}
+
+# Assemble group txn
+proc ::AlgorandGoal::AssembleGroup { INPUT_GROUP OUTPUT_GROUP } {
+    if { [ catch {
+        spawn goal clerk group -i $INPUT_GROUP -o $OUTPUT_GROUP
+	expect {
+            timeout { ::AlgorandGoal::Abort "Failed to assemble group transaction"  }
```

Comment:
Could you change the error text to indicate that the error was triggered by a timeout ?